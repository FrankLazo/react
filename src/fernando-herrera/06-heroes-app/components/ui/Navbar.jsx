import { useContext, useState } from 'react';
import { Link, NavLink, useHistory } from 'react-router-dom';
import AuthContext from '../../auth/AuthContext';
import { types } from '../../types/types';

const Navbar = () =>
{
	const history = useHistory();
	const [burger, setBurger] = useState(false);
	const { user: { user }, dispatch } = useContext(AuthContext);

	const handleBurger = () =>
	{
		setBurger(!burger);
	};

	const handleLogout = () =>
	{
		dispatch({
			type: types.logout,
		});

		history.replace('./login');
	};

    return (
        <nav className="navbar">
            <div className="navbar-brand">
				<Link to="/" className="navbar-item has-text-weight-bold">Asociaciones</Link>

				<button className={`navbar-burger ${ burger && 'is-active' }`} onClick={ handleBurger }>
					<span aria-hidden="true"></span>
					<span aria-hidden="true"></span>
					<span aria-hidden="true"></span>
				</button>
			</div>

            <div className={`navbar-menu ${ burger && 'is-active' }`}>
				<ul className="navbar-start">
					<li className="navbar-item">
						<NavLink exact activeClassName="has-text-primary" to="/marvel">Marvel</NavLink>
					</li>
					<li className="navbar-item">
						<NavLink exact activeClassName="has-text-primary" to="/dc">DC</NavLink>
					</li>
					<li className="navbar-item">
						<NavLink exact activeClassName="has-text-primary" to="/search">Search</NavLink>
					</li>
				</ul>

				<ul className="navbar-end">
                    <li className="navbar-item">
						<span className="has-text-weight-medium">Hello { user }!</span>
					</li>
                    <li className="navbar-item">
						<div className="buttons">
							<button
								className="button is-primary"
								onClick={ handleLogout }
								>
								Logout
							</button>
						</div>
					</li>
                </ul>
            </div>
        </nav>
    );
};

export default Navbar;
