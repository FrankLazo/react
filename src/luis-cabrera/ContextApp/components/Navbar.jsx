import { useContext, useState } from 'react';
import UserContext from '../contexts/UserContext';

const Navbar = () => {
	const [isShowedMenu, setIsShowedMenu] = useState(false);
	const { user, login, logout } = useContext(UserContext);

	return (
		<nav className="navbar is-success">
			<div className="navbar-brand">
				<div
					className="navbar-burger"
					onClick={ () => setIsShowedMenu(!isShowedMenu) }>
					<span aria-hidden="true"></span>
					<span aria-hidden="true"></span>
					<span aria-hidden="true"></span>
				</div>
			</div>

			<div className={`navbar-menu ${ isShowedMenu && 'is-active' }`}>
				<div className="navbar-start">
					<div className="navbar-item">
						<p className="is-size-4">{ user ? `Hola ${ user.name }!` : 'Bienvenid@' }</p>
					</div>
				</div>

				<div className="navbar-end">
					<div className="navbar-item">
						<div className="buttons">
							{
								user
								? (
									<button
										className="button is-danger"
										onClick={ logout }
										>
										Cerrar sesión
									</button>
								)
								: (
									<button
										className="button is-danger"
										onClick={ login }
										>
										Iniciar sesión
									</button>
								)
							}
						</div>
					</div>
				</div>
			</div>
		</nav>
	);
}

export default Navbar;
