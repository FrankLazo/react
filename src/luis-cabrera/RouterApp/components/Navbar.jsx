import { useState } from 'react';
import { NavLink } from 'react-router-dom';
import useAuth from '../auth/useAuth';

const Navbar = () =>
{
	const auth = useAuth();
	const [showMenu, setShowMenu] = useState(false);

	return (
		<nav className="navbar is-dark" role="navigation" aria-label="main navigation">
			<div className="navbar-brand">
				<div
					className={`navbar-burger ${ showMenu && 'is-active' }`}
					onClick={() => setShowMenu(!showMenu)}
					>
					<span aria-hidden="true"></span>
					<span aria-hidden="true"></span>
					<span aria-hidden="true"></span>
				</div>
			</div>

			<div className={`navbar-menu ${ showMenu && 'is-active' }`}>
				<ul className="navbar-start">
					<li className="navbar-item">
						<NavLink exact to="/"
						className={`${ showMenu ? 'has-text-dark' : 'has-text-light' }`}
						activeClassName="has-text-primary">Home</NavLink>
					</li>
					<li className="navbar-item">
						<NavLink to="/about"
						className={`${ showMenu ? 'has-text-dark' : 'has-text-light' }`}
						activeClassName="has-text-primary">About</NavLink>
					</li>
					<li className="navbar-item">
						<NavLink to="/contact"
						className={`${ showMenu ? 'has-text-dark' : 'has-text-light' }`}
						activeClassName="has-text-primary">Contact</NavLink>
					</li>
					<li className="navbar-item">
						<NavLink to="/categories"
						className={`${ showMenu ? 'has-text-dark' : 'has-text-light' }`}
						activeClassName="has-text-primary">Categories</NavLink>
					</li>

				{
					auth.isLogged() && (
						<>
							<li className="navbar-item">
								<NavLink to="/dashboard"
								className={`${ showMenu ? 'has-text-dark' : 'has-text-light' }`}
								activeClassName="has-text-primary">Dashboard</NavLink>
							</li>
							<li className="navbar-item">
								<NavLink to="/payments"
								className={`${ showMenu ? 'has-text-dark' : 'has-text-light' }`}
								activeClassName="has-text-primary">Payments</NavLink>
							</li>
						</>
					)
				}
				</ul>

				<ul className="navbar-end">
				{
					!auth.isLogged() && (
						<>
							<li className="navbar-item">
								<NavLink to="/login"
								className={`${ showMenu ? 'has-text-dark' : 'has-text-light' }`}
								activeClassName="has-text-primary">Login</NavLink>
							</li>
							<li className="navbar-item">
								<NavLink to="/register"
								className={`${ showMenu ? 'has-text-dark' : 'has-text-light' }`}
								activeClassName="has-text-primary">Register</NavLink>
							</li>
						</>
					)
				}

				{
					auth.isLogged() && (
						<li className="navbar-item">
							<div className="buttons">
								<button className="button is-primary" onClick={auth.logout}>
									Logout
								</button>
							</div>
						</li>
					)
				}
				</ul>
			</div>
		</nav>
	);
}

export default Navbar;
