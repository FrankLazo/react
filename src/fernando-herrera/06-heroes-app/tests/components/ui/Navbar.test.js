import '@testing-library/jest-dom'
import { mount } from 'enzyme';
import { MemoryRouter, Router } from 'react-router-dom';
import AuthContext from '../../../auth/AuthContext';
import Navbar from '../../../components/ui/Navbar';
import { types } from '../../../types/types';

describe('<Navbar /> tests', () => {
	const historyMock = {
		location: {},
		push: jest.fn(),
		listen: jest.fn(),
		replace: jest.fn(),
		createHref: jest.fn(),
	};

	const contextValue = {
		user: {
			logged: false,
			user: 'Frank',
		},
		dispatch: jest.fn(),
	};

	const wrapper = mount(
		<AuthContext.Provider value={ contextValue }>
			<MemoryRouter>
				<Router history={ historyMock }>
					<Navbar />
				</Router>
			</MemoryRouter>
		</AuthContext.Provider>
	);

	afterEach(() => {
		jest.clearAllMocks();
	});

	test('should show the right component', () => {
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('.has-text-weight-medium').text().trim()).toBe('Hello Frank!');
	});

	test('should call logout and history', () => {
		const button = wrapper.find('.button');
		button.simulate('click');

		expect(contextValue.dispatch).toHaveBeenCalledWith({ type: types.logout });
		expect(historyMock.replace).toHaveBeenCalledWith('./login');
	});
});
