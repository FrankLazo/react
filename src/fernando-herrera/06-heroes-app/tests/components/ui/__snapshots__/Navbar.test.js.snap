// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> tests should show the right component 1`] = `
<nav
  className="navbar"
>
  <div
    className="navbar-brand"
  >
    <a
      className="navbar-item has-text-weight-bold"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <button
      className="navbar-burger false"
      onClick={[Function]}
    >
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
    </button>
  </div>
  <div
    className="navbar-menu false"
  >
    <ul
      className="navbar-start"
    >
      <li
        className="navbar-item"
      >
        <a
          aria-current={null}
          onClick={[Function]}
        >
          Marvel
        </a>
      </li>
      <li
        className="navbar-item"
      >
        <a
          aria-current={null}
          onClick={[Function]}
        >
          DC
        </a>
      </li>
      <li
        className="navbar-item"
      >
        <a
          aria-current={null}
          onClick={[Function]}
        >
          Search
        </a>
      </li>
    </ul>
    <ul
      className="navbar-end"
    >
      <li
        className="navbar-item"
      >
        <span
          className="has-text-weight-medium"
        >
          Hello 
          Frank
          !
        </span>
      </li>
      <li
        className="navbar-item"
      >
        <div
          className="buttons"
        >
          <button
            className="button is-primary"
            onClick={[Function]}
          >
            Logout
          </button>
        </div>
      </li>
    </ul>
  </div>
</nav>
`;
